File: examples/src/main/java/jcdc/pluginfactory/betterexamples/TreeDelogger.java
package jcdc.pluginfactory.betterexamples;

import jcdc.pluginfactory.java.BetterJavaPlugin;

import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;

public class TreeDelogger extends BetterJavaPlugin {
  public TreeDelogger() {
    listeners.add(new Listener() {
      @EventHandler public void onBlockBreak(BlockBreakEvent event) {
        Block current = event.getBlock();
        while (current.getType() == Material.LOG) {
          erase(current);
          current = blockAbove(current);
        }
      }
    });
  }
}

File: examples/TreeDelogger/TreeDelogger.scala
package com.joshcough.minecraft.examples

import org.bukkit.Material._
import com.joshcough.minecraft.Listeners
import Listeners._

class TreeDelogger extends ListeningFor(OnBlockBreak { (b,_,_) =>
  if (b isA LOG) for (b <- b.andBlocksAbove.takeWhile(_ isA LOG)) b.erase
})
